app/src/main/java/chat/rocket/android/profile/ui/ProfileFragment.kt

```diff
@@ -1,30 +1,63 @@
 package chat.rocket.android.profile.ui
 
 import DrawableHelper
+import android.Manifest
+import android.app.Activity.RESULT_OK
+import android.app.AlertDialog
+import android.content.DialogInterface
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.graphics.Bitmap
 import android.os.Build
 import android.os.Bundle
+import android.os.Environment
+import android.provider.MediaStore
+import android.support.v4.app.ActivityCompat
 import android.support.v4.app.Fragment
+import android.support.v4.content.ContextCompat
+import android.support.v4.content.PermissionChecker.PERMISSION_GRANTED
 import android.support.v7.view.ActionMode
+import android.util.Log
 import android.view.*
+import android.widget.ListAdapter
+import android.widget.Toast
 import chat.rocket.android.R
 import chat.rocket.android.main.ui.MainActivity
 import chat.rocket.android.profile.presentation.ProfilePresenter
 import chat.rocket.android.profile.presentation.ProfileView
 import chat.rocket.android.util.extensions.*
 import dagger.android.support.AndroidSupportInjection
 import io.reactivex.rxkotlin.Observables
+import io.reactivex.subjects.PublishSubject
+import io.reactivex.subjects.Subject
 import kotlinx.android.synthetic.main.app_bar.*
 import kotlinx.android.synthetic.main.avatar_profile.*
 import kotlinx.android.synthetic.main.fragment_profile.*
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.util.*
 import javax.inject.Inject
 
+
 class ProfileFragment : Fragment(), ProfileView, ActionMode.Callback {
-    @Inject lateinit var presenter: ProfilePresenter
+    @Inject
+    lateinit var presenter: ProfilePresenter
     private lateinit var currentName: String
     private lateinit var currentUsername: String
     private lateinit var currentEmail: String
     private lateinit var currentAvatar: String
     private var actionMode: ActionMode? = null
+    private var avatarImage: File? = null
+    private var isAvatarChanged = false
+    //request codes
+    private var CHOOSE_PICKER_MODE = 193
+    private val CAMERA_REQUEST_CODE = 108
+    private val READ_STORAGE_REQUEST_CODE = 109
+
+    private var mObservable: Subject<Boolean> = PublishSubject.create()
```

Comment:
We aren't using the Hungarian notation on the project (code style). Could you change it?